### DISCUSSION TITLE
Help to use Transmit

### DISCUSSION THREAD

**Discussion by shazzsamed** [adonisjs/core#4728] [Help]
I am very curious to find out if anyone has used Transmit which is an SSE module in adonis js.
I using adonis js to build the backend of my application. And for features like push notifications i thought of using Transmit as an option. But i am stuck with the documentation. I am using Vue js for the client side. If someone has used this before please do help out. It would be great if someone would walk me through a simple setup of this

**Comment by kitro** [comment]
Hi @shazzsamed!,
The documentation of Transmit is straightforward, can you please describe what part you are blocked in. so we can help.
The Transmit is for real-time in-app communication between the frontend and the backend. The push notifications is differer from SSE or WS. if you want to integrate a push notification feature in the most case you can use [FCM (Firebase cloud messaging)](https://firebase.google.com/products/cloud-messaging), Amazon SNS, or any other provider.

**Comment by shazzsamed** [comment]
I dont quite understand the use of it i'll attach the the code snippets clearly where i've worked

### Server-Side
```
/start/routes.ts
import transmit from '@adonisjs/transmit/services/main'

transmit.registerRoutes()
```

```
/config/transmit.ts
import { defineConfig } from '@adonisjs/transmit'

export default defineConfig({
  pingInterval: false,
  transport: null,
})

```

```
/app/controller/comment_controller.ts
import transmit from '@adonisjs/transmit/services/main'
export default class CommentsController {
  public async createComment(ctx: HttpContext) {
    //code
    transmit.broadcast('global', { message: 'Hello' })
    //code
}
```
### Client-Side (Vue-js)
```
/client/app/transmitClinet.ts
import { Transmit } from "@adonisjs/transmit-client";

export const transmit = new Transmit({
  baseUrl: window.location.origin,
});
```

```
/layouts/default.vue
<template>
//html files
</template>
<script>
console.log("Creating subscription...");
try {
  const subscription = transmit.subscription("global"); 
  await subscription.create();
  console.log("Subscription created");

  subscription.onMessage((data) => {
    console.log(data);
  });
} catch (error) {
  console.error("Subscription error:", error);
}
</script>
```
logs in client console
![image](https://github.com/user-attachments/assets/be7adec0-142b-4a59-814e-0ae352e017e8)

leave alone the message being displayed the subscription is itself not being created

**Reply by RomainLanz** [reply]
Can you please share the request from your network tab?
